tests/tests/operation_time_tests.cpp

```diff
@@ -23,6 +23,131 @@ using namespace steemit::protocol;
+      }
+
+      // authors all write in the same block, but Bob declines payout
+      for( const auto& author : authors )
+      {
+         signed_transaction tx;
+         comment_operation com;
+         com.author = author.name;
+         com.permlink = "mypost";
+         com.parent_author = STEEMIT_ROOT_POST_PARENT;
+         com.parent_permlink = "test";
+         com.title = "Hello from "+author.name;
+         com.body = "Hello, my name is "+author.name;
+         tx.operations.push_back( com );
+
+         if( author.max_accepted_payout.valid() )
+         {
+            comment_options_operation copt;
+            copt.author = com.author;
+            copt.permlink = com.permlink;
+            copt.max_accepted_payout = *(author.max_accepted_payout);
+            tx.operations.push_back( copt );
+         }
+
+         tx.set_expiration( db.head_block_time() + STEEMIT_MAX_TIME_UNTIL_EXPIRATION );
+         tx.sign( author.private_key, db.get_chain_id() );
+         db.push_transaction( tx, 0 );
+      }
+
+      generate_blocks(1);
+
+      // voters all vote in the same block with the same stake
+      for( const auto& voter : voters )
+      {
+         signed_transaction tx;
+         vote_operation vote;
+         vote.voter = voter.name;
+         vote.author = voter.favorite_author;
+         vote.permlink = "mypost";
+         vote.weight = STEEMIT_100_PERCENT;
+         tx.operations.push_back( vote );
+         tx.set_expiration( db.head_block_time() + STEEMIT_MAX_TIME_UNTIL_EXPIRATION );
+         tx.sign( voter.private_key, db.get_chain_id() );
+         db.push_transaction( tx, 0 );
+      }
+
+      auto reward_steem = db.get_dynamic_global_properties().total_reward_fund_steem;
+      auto total_rshares2 = db.get_dynamic_global_properties().total_reward_shares2;
+
+      // generate a few blocks to seed the reward fund
+      generate_blocks(10);
+      ilog( "dgpo: ${dgpo}", ("dgpo", db.get_dynamic_global_properties()) );
+
+      generate_blocks( db.get_comment( "alice", string( "mypost" ) ).cashout_time, true );
+      for( const auto& author : authors )
+      {
+         const account_object& a = db.get_account(author.name);
+         ilog( "${n} : ${steem} ${sbd}", ("n", author.name)("steem", a.balance)("sbd", a.sbd_balance) );
+      }
+      for( const auto& voter : voters )
+      {
+         const account_object& a = db.get_account(voter.name);
+         ilog( "${n} : ${steem} ${sbd}", ("n", voter.name)("steem", a.balance)("sbd", a.sbd_balance) );
+      }
+
+      const account_object& alice_account = db.get_account("alice");
+      const account_object& bob_account   = db.get_account("bob");
+      const account_object& dave_account  = db.get_account("dave");
+
+      BOOST_CHECK( alice_account.sbd_balance == ASSET( "10720.000 TBD" ) );
+      BOOST_CHECK( bob_account.sbd_balance == ASSET( "0.000 TBD" ) );
+      BOOST_CHECK( dave_account.sbd_balance == alice_account.sbd_balance );
```

Comment:
What if the comments have different rshares? The test should prove that they are paid out equal Steem per rshares.